index.html

```diff
@@ -138,233 +137,254 @@
-
-          if (code == 13) {
-            e.preventDefault();
-
-            if ($(this).val()) {
-              loadPaste($(this).val());
-            } else {
-              alert("No ID!");
-            }
-          }
-        });
-
-        $("#uuid").keypress(function(e) {
-          var code = e.keyCode || e.which;
-
-          if (code == 13) {
-            e.preventDefault();
-
-            if ($(this).val()) {
-              updatePaste($(this).val());
-            } else {
-              alert("No UUID!");
-            }
-          }
-        });
-
-        $("#expire").on("input", function(e) {
-          var v = $(this).val();
-
-          if (v == 1) {
-            $("#hours").text("hour");
-          } else if (v == 0) {
-            $(this).val("");
-            $("#hours").text("hours");
-          } else {
-            $("#hours").text("hours");
-          }
-        });
-      });
+    <div id="paste_error" hidden="true">
+      <ul id="error_list"></ul>
+    </div>
+
+    <script>
+     $(document).ready(function() {
+       document.title = "pb";
+
+       $("a").attr("title", "Opens in a new tab");
+
+       $("button").addClass("btn btn-default");
+
+       // button colors
+       $("button#delete").addClass("btn-danger");
+       $("button#submit").addClass("btn-primary");
+       $("button#clear").addClass("btn-danger");
+
+
+       $("#file").hide();
+
+       $("#clear").click(function(e) {
+         e.preventDefault();
+
+         $("#text").val("");
+         $("#private").prop("checked", false);
+         $("#expire").val("");
+         $("#file").val("");
+         $("#filename").text("");
+         $("#pasteid").val("");
+         $("#uuid").val("");
+
+         $("#paste_response").hide();
+         $("#text").focus();
+       });
+
+       function getData() {
+         data = {
+           c: $("#text").val()
+         }
+
+         if ($("#private").prop("checked")) {
+           data.p = 1;
+         }
+
+         if ($("#expire").val() > 0) {
+           data.s = $("#expire").val() * 3600;
+         }
+
+         return data;
+       }
+
+       function displayResponse(response) {
+         $("#paste_error").hide();
+         $("#paste_response").show();
+
+         $("#response_url").attr("href", response.url);
+         $("#response_url").text(response.url);
+         $("#response_status").text(response.status);
+         $("#response_uuid").text(response.uuid);
+         $("#pasteid").val(response.short);
+
+         if (response.uuid) {
+           $("#uuid").val(response.uuid);
+         }
+       }
+
+       // text -> JSON -> html
+       function jsonTextToHtmlList (text) {
+         var json = JSON.parse(text);
+         return Object.keys(json).map(function (key) {
+           return "<li>" + key + " : " + json[key] + "</li>"
+         }).join("");
+       }
+
+       function displayError(text) {
+         $("#error_list").html(jsonTextToHtmlList(text));
```

Comment:
Success is determined by [`status >= 200 && status < 300 || status === 304`](https://github.com/jquery/jquery/blob/master/src/ajax.js).

While all *pb* non-`GET` API responses are going to be valid json if you asked for json, [other things](https://github.com/ptpb/pb/issues/135) like Nginx, Varnish, Flask, and Werkzeug can give you responses too, which are usually going to be html.

In that case jquery returns you a not-object, which you pass to `JSON.parse`, which then throws `SyntaxError`.

I suggest ignoring the existence of `success` and `fail`, using `done` instead, and looking at `status` values to determine how to color this. If you get a `Content-Type: text/html` response, just show the HTTP status line or something.